ladybug/epw.py

```diff
@@ -87,16 +109,21 @@ def file_name(self):
     @file_path.setter
     def file_path(self, epw_file_path):
         """Get path to epw file."""
-        self._file_path = os.path.normpath(epw_file_path)
+        if epw_file_path is not None:
+            self._file_path = os.path.normpath(epw_file_path)
 
-        if not os.path.isfile(self._file_path):
-            raise ValueError(
-                'Cannot find an epw file at {}'.format(self._file_path))
+            if not os.path.isfile(self._file_path):
+                raise ValueError(
+                    'Cannot find an epw file at {}'.format(self._file_path))
 
-        if not epw_file_path.lower().endswith('epw'):
-            raise TypeError(epw_file_path + ' is not an .epw file.')
+            if not epw_file_path.lower().endswith('epw'):
+                raise TypeError(epw_file_path + ' is not an .epw file.')
 
-        self._folder, self._file_name = os.path.split(self.file_path)
+            self._folder, self._file_name = os.path.split(self.file_path)
+        else:
```

Comment:
@AntoineDao, If the filepath is not set correctly we should raise an error, shouldn't we? Are you thinking about creating an EPW object from the data in database?